src/focus.js

```diff
@@ -0,0 +1,37 @@
+﻿import {customAttribute} from 'aurelia-templating';
+import {inject} from 'aurelia-dependency-injection';
+
+var afterAttachedSetFocusDelay = 50;
+
+@customAttribute('focus')
+@inject(Element)
+export class Focus {
+    constructor(element) {
+        this.element = element;
+
+        this.focusListener = (e => this.value = true).bind(this);
+        this.blurListener = (e => this.value = false).bind(this);
+    }
+
+    valueChanged(newValue) {
+        if (newValue) {
+            this.element.focus();
+        } else {
+            this.element.blur();
+        }
+    }
+
+    attached() {
+        this.element.addEventListener('focus', this.focusListener);
+        this.element.addEventListener('blur', this.blurListener);
+
+        if (this.value) {
+            setTimeout(() => this.element.focus(), afterAttachedSetFocusDelay);
```

Comment:
If the value is changed within `afterAttachedSetFocusDelay` of attaching then this will incorrectly focus the element. I think we need to either store the timeout ID and cancel it in `valueChanged`, or check the value again inside the callback.